internal/host/docker_unix.go

```diff
@@ -55,9 +52,19 @@ func (d Docker) Run(ctx context.Context, cmd host.Cmd) (host.WaitStatus, error)
 	if cmd.Stdin != nil {
 		args = append(args, "--interactive")
 	}
-	args = append(args, []string{"--user", d.User}...)
+
+	parts := strings.Split(d.ConnectionString, "@")
+
+	if len(parts) < 2 {
+		return host.WaitStatus{}, fmt.Errorf("invalid connection string format")
+	}
+
+	dockerConnectionUser := parts[0]
+	dockerConnectionContainer := parts[1]
```

Comment:
I think we may be missing the case where we just get the container (without `user@` prefix), and the error:

```suggestion
	var dockerConnectionUser, dockerConnectionContainer string
	switch len(parts) {
	case 1:
		dockerConnectionUser = "0:0"
		dockerConnectionContainer = parts[0]
	case 2:
		dockerConnectionUser = parts[0]
		dockerConnectionContainer = parts[1]
	default:
		return host.WaitStatus{}, fmt.Errorf("invalid connection string format: must match [<name|uid>[:<group|gid>]@<image>, got %s", d.ConnectionString)
	}
```